staging/src/k8s.io/component-base/config/v1alpha1/defaults.go

```diff
@@ -44,7 +44,7 @@ func RecommendedDefaultLeaderElectionConfiguration(obj *LeaderElectionConfigurat
 		obj.RetryPeriod = metav1.Duration{Duration: 2 * time.Second}
 	}
 	if obj.ResourceLock == "" {
-		// TODO: Migrate to LeaseLock.
+		// TODO(#80289): Migrate to LeaseLock when graduating to v1beta1.
```

Comment:
Going straight from Endpoints -> Lease isn't safe, right? We have to go Endpoints -> EndpointsLeases -> Lease over three releases, right?

Reply:
Separately, I'm not sure we can know when it's safe to migrate to LeaseLock. With other components, we know their rollout cadence and skew support, so we can go endpoints -> endpointsleases -> leases over three versions. With this helper method, we have no idea what the consuming component's release schedule and skew support is.

Reply:
Yes - going Endpoints->Leases isn't safe.

Reply:
done